controllers/validation.go

```diff
@@ -108,7 +116,30 @@ func (r *ClusterGroupUpgradeReconciler) extractOpenshiftImagePlatformFromPolicie
 func (r *ClusterGroupUpgradeReconciler) validateOpenshiftUpgradeVersion(
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policies []*unstructured.Unstructured) error {
 
-	_, err := r.extractOpenshiftImagePlatformFromPolicies(policies)
+	versionInfo, err := extractOCPVersionInfoFromPolicies(policies)
+
+	if err == nil {
```

Comment:
With the current construct, we're not updating the CGU status with the first error - we might actually be getting a couple of errors, but we're only saving the last one in the status. It might be better to move this `if err == nill` in a function and `return err` for each error case? Just an idea.

Reply:
Not sure what you mean. If something happens in the extract function, we skip this and set status at the end.

Reply:
I see what you mean now. The branches in this if block are actually mutually exclusive. Let me update the last else block. The image check there is actually redundant. The query function already does that.

Reply:
Ah, nevermind - yes, it looks good.